src/test/spec/unified_runner/test_runner.rs

```diff
@@ -462,62 +436,113 @@ impl TestRunner {
                 TestFileEntity::Bucket(_) => {
                     panic!("GridFS not implemented");
                 }
+                TestFileEntity::Thread(thread) => {
+                    let (sender, mut receiver) = mpsc::unbounded_channel::<ThreadMessage>();
+                    let runner = self.clone();
+                    let d = description.as_ref().to_string();
+                    let id = thread.id.clone();
+                    runtime::execute(async move {
+                        while let Some(msg) = receiver.recv().await {
+                            match msg {
+                                ThreadMessage::ExecuteOperation(op) => {
+                                    op.execute(runner.clone(), d.as_str()).await;
+                                }
+                                ThreadMessage::Stop(sender) => {
+                                    sender.send(Ok(())).unwrap_or_else(|_| {
```

Comment:
idk if I follow the error case here that we're panicking on here... if I understand correctly, we only receive `ThreadMessage::Stop` here once `wait()` has been called on a `ThreadEntity`, which looks to only occur via a `waitForThread` operation being executed. 

from the docs it looks `oneshot::Sender::send` only fails if the receiver has already been dropped. maybe this can happen if we for some reason kick off but don't finish awaiting the `waitForThread` operation, so `ThreadEntity.wait()` doesn't actually finish executing and the receiver is dropped? or how were you imagining we end up in this state?